### PR TITLE
Dockerfile and basic CI for Go relay maintainer

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Depends on:  #762

This pull request introduces the `Dockerfile` for the Go relay maintainer along with a basic `build-test-publish` CI job.

### CODE REVIEW

File: .github/workflows/relay.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Run Go tests
+        run: |
+          docker run \
+            --workdir /go/src/github.com/keep-network/tbtc/relay \
+            go-build-env \
+            gotestsum
+
+      - name: Login to Google Container Registry
+        if:  |
+          startsWith(github.ref, 'refs/heads/releases/')
```
Comment (michalinacienciala) [inline .github/workflows/relay.yml:58]: After [this discussion in Flowdock](https://www.flowdock.com/app/cardforcoin/tech/threads/sqQIWXEcGU8O2Gf-mMqgsfmT6wr) I updated `keep-core` and `keep-ecdsa` workflows to execute steps publishing client only when workflow runs due to merge to master. Changes of the config aren't yet in the master, they're waiting for review in https://github.com/keep-network/keep-core/pull/2378 and https://github.com/keep-network/keep-ecdsa/pull/738. Do we want same rules for publishing of relay maintainer (that wasn't discussed in FD)? If yes, we can apply same config here.
```suggestion
          github.ref == 'refs/heads/master'
            && (github.event_name == 'push'
            || github.event_name == 'workflow_dispatch')
```

File: .github/workflows/relay.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          context: ./relay
+          # GCR image should be named according to following convention:
+          # HOSTNAME/PROJECT-ID/IMAGE:TAG
+          # We don't use TAG yet.
+          tags: ${{ secrets.GCR_REGISTRY_URL }}/${{ env.GOOGLE_PROJECT_ID }}/${{ env.IMAGE_NAME }}
+          labels: revision=${{ github.sha }}
+          build-args: |
+            REVISION=${{ github.sha }}
+          push: |
+            ${{ startsWith(github.ref, 'refs/heads/releases/') }}
```
Comment (michalinacienciala) [inline .github/workflows/relay.yml:80]: Same thing as with the comment for line 58:
```suggestion
          push: |
            ${{ github.ref == 'refs/heads/master'
              && (github.event_name == 'push'
              || github.event_name == 'workflow_dispatch') }}
```

File: .github/workflows/relay.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-relay-cache
+          key: ${{ runner.os }}-buildx-relay-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-relay-
+
+      - run: sudo df -h
+
```
Comment (michalinacienciala) [inline .github/workflows/relay.yml:38]: We no longer need to monitor the disk space, so those line can be removed (I removed them in `keep-core` and `keep-ecdsa` in recent PRs - not yet merged).

**Review by michalinacienciala — COMMENTED** [review]
I left some comments regarding ci job. I also took a look at Dockerfile, but I'm not an expert on it, so I'm leaving it for review for somebody more experienced.

File: .github/workflows/relay.yml (hunk 1 above)
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:58]: Yep I think it's a good idea. Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

File: .github/workflows/relay.yml (hunk 2 above)
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:80]: Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

File: .github/workflows/relay.yml (hunk 3 above)
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:38]: Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

**Comment by lukasz-zimnoch** [comment]
@michalinacienciala ready for another look!

**Review by michalinacienciala — APPROVED** [review]

**Review by tomaszslabon — APPROVED** [review]

**Pull request merged**